src/counter.rs

```diff
@@ -64,12 +64,10 @@ impl<P: Atomic> GenericCounter<P> {
     ///
     /// # Panics
     ///
-    /// Panics if the value is < 0.
+    /// Panics in debug build if the value is < 0.
     #[inline]
     pub fn inc_by(&self, v: P::T) {
-        if v < P::T::from_i64(0) {
-            panic!("counter cannot inc negative values")
-        }
+        debug_assert!(v >= P::T::from_i64(0));
```

Comment:
if less than 0, I prefer returning directly, not affecting the counter.

Reply:
I think overflow is better. Once there is a value less than 0, both let it overflow and ignore it lead to wrong result. Nobody knows what's the correct way to handle it. However let it overflow is the cheapest way to notify someone that something is wrong here.

Reply:
If overflow, we may still add some incorrect data, and this may cause the metric wrong but sometimes we don't know it. Maybe return directly and keep the counter unchanged can help us figure the problem more easily.

Reply:
> ...sometimes we don't know it

If the counter can overflow in a long running system, it will probably overflow again and again.
  
> ...keep the counter unchanged...

So the counter needs another field to keep the state?

Reply:
@siddontang @BusyJay Are there any final conclusions?

Reply:
I still think overflow is the simplest and "rusty" solution.

Reply:
I agree with @BusyJay

Reply:
ok, you can go on